### DISCUSSION TITLE
Help with booting Windows 10

### DISCUSSION THREAD

**Discussion by rbhorse** [bochs-emu/Bochs#219] [Q&A]
I am trying to get windows 10 working on bochs.
Can someone who has succeeded in the task share their bochsrc? I have a basic bochsrc at the moment the moment is capable of booting tinycore linux but I'm missing out on specific configuration for booting windows 10.

At the moment, using a .vdi file converted to sparse (because booting directly from .vdi ends up with a bios error) fails to boot with 'PROCESS1_INITIALIZATION_FAILED' and using the Windows 10 installation iso fails with 'IRQ_LESS_THAN_OR_EQUAL'

**Comment by Vort** [comment]
I did not tried Win 10 yet, but I saw similar problems with conversion from VirtualBox.
You may try changing `chipset=i440fx` to `i440bx`.

**Comment by fysnet** [comment]
Here is a stripped down version of a bochsrc.txt file I use to boot Tiny Windows 10.
```
config_interface: win32config
romimage: file=C:/bochs/bochs/bios/BIOS-bochs-latest
cpu: model=corei7_sandy_bridge_2600k
cpu: count=1, ips=325000000, reset_on_triple_fault=1, ignore_bad_msrs=1
cpu: cpuid_limit_winnt=0
clock: sync=none, time0="Thu Mar  09 00:00:00 2023"
memory: guest=1024, host=1024
vgaromimage: file=C:/bochs/bochs/bios/VGABIOS-lgpl-latest-cirrus
vga: extension=cirrus, update_freq=5
ata0: enabled=1, ioaddr1=0x1f0, ioaddr2=0x3f0, irq=14
ata1: enabled=1, ioaddr1=0x170, ioaddr2=0x370, irq=15
ata2: enabled=0, ioaddr1=0x1e8, ioaddr2=0x3e0, irq=11
ata3: enabled=0, ioaddr1=0x168, ioaddr2=0x360, irq=9
ata0-master: type=disk, path="tiny10.img", model="tiny10.img", translation=none
boot: disk
log: log.txt
panic: action=ask
error: action=report, cpu0=ignore
info: action=report
debug: action=ignore
parport1: enabled=1, file="parport1.txt"
mouse: enabled=0, type=imps2
private_colormap: enabled=0
pci: enabled=1, chipset=i440fx, slot1=cirrus
port_e9_hack: enabled=0
print_timestamps: enabled=1
```

**Comment by stlintel** [comment]
I see it boots with no error with cpu: model=corei7_sandy_bridge_2600k or even with cpu: model=corei7_haswell_4770 but crash with any newer CPU model. 
After initial triage it seems like ADCX/ADOX emulation creating an issue and causing PROCESS1_INITIALIZATION_FAILED BSOD.
Investigating ...

EDIT: Found and fixed bug in ADOX instruction. Now Win10 boots with any CPU configuration.

**Comment by stlintel** [comment]
EDIT: with tigerlake and sapphire_rapids boot fails on BSOD PAGE_FAULT_IN_NON_PAGED_AREA
Seems like CET model has yet a lot to improve.

**Comment by stlintel** [comment]
Found two more XSAVE/XRSTOR related issues during Win10 boot. Fixed in git,
Now Win10 behaves the same with ANY CPU model includes ones with CET support like tigerlake and  sapphire_rapids.
